source/common/common/matchers.cc

```diff
@@ -113,6 +115,8 @@ LowerCaseStringMatcher::toLowerCase(const envoy::type::matcher::v3::StringMatche
     lowercase.set_hidden_envoy_deprecated_regex(
         StringUtil::toLower(matcher.hidden_envoy_deprecated_regex()));
     break;
+  case envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kSafeRegex:
```

Comment:
Regular regexes are still lower cased, is this an issue?

Reply:
It seems like a number of negated character classes will not work as expected. For example `\D` (non-digit), `\S` (non-whitespace), and `\W` (non-word).

It may be too big a change in the short term, but I think this whole LowerCaseStringMatcher class should be removed. It seems only to be used in ext_authz for header names. I think that filter should eventually just document that header name matchers must match lower-case names.

Reply:
Added 8b1e8e5 to remove `LowerCaseStringMatcher`.